include/engine/guidance/route_step.hpp

```diff
@@ -144,6 +146,14 @@ inline RouteStep &RouteStep::AddInFront(const RouteStep &preceeding_step)
     return *this;
 }
 
+// checks whether we can actually merge two route steps:
+// They neeed to have the same mode, because a route step
+// can only have one mode.
+inline bool RouteStep::IsCompatible(const RouteStep &other) const
+{
+    return mode == other.mode;
+}
```

Comment:
We have a similar concept for graph compression, I didn't find anything explicit for guidance.